src/coreclr/jit/lclvars.cpp

```diff
@@ -2634,6 +2634,28 @@ bool Compiler::StructPromotionHelper::IsArmHfaParameter(unsigned lclNum)
     return hfaType != CORINFO_HFA_ELEM_NONE;
 }
 
+//--------------------------------------------------------------------------------------------
+// IsSysVMultiRegType - Check if a type is one that could be passed in 2
+// registers in some cases.
+// This is a quirk to match old promotion behavior.
+//
+// Arguments:
+//   lclNum - The local
+//
+// Return value:
+//   True if it sometimes may be passed in two registers.
+//
+bool Compiler::StructPromotionHelper::IsSysVMultiRegType(ClassLayout* layout)
+{
+#ifdef UNIX_AMD64_ABI
```

Comment:
not the case for arm64 (sysv/win)?

Reply:
Not according to diffs... To be honest I did not spend too much time looking at why this turns out to be a good heuristic for SysV, since I hope to remove all of this in this or next release.